extension/lib/shopgate/customer/get.js

```diff
@@ -0,0 +1,76 @@
+const BigCommerceCustomerRepository = require('../../bigcommerce/CustomerRepository')
+const { decorateError } = require('../logDecorator')
+const ClientRequestError = require('./errors/ClientRequestError')
+const UnknownError = require('./errors/UnknownError')
+
+let customerRepo = null
+/**
+ * @param {PipelineContext} context
+ * @param {string} email
+ * @return {Promise<LoginResponse>}
+ */
+module.exports = async function getCustomer (context, email) {
+  if (!customerRepo) {
+    customerRepo = BigCommerceCustomerRepository.create(
+      context.config.clientId,
+      context.config.accessToken,
+      context.config.storeHash
+    )
+  }
+
+  let customer = null
+  try {
+    customer = await customerRepo.getCustomerByEmail(email)
+  } catch (e) {
+    // Try parsing the (potential) underlying api error
+    const errorMessageMatch = e.message.match(/({.+})/)
+
+    if (!errorMessageMatch) {
+      context.log.error(decorateError(e), 'Error in login process.')
+      throw new UnknownError()
+    }
+
+    let parsed
+    try {
+      parsed = JSON.parse(errorMessageMatch[1])
+    } catch (unparseable) {
+      context.log.error(
+        decorateError(unparseable),
+        'Unable to process the error from BigC api'
+      )
+    }
+
+    if (!parsed) {
+      context.log.error(decorateError(e), 'Empty error message')
+      throw new UnknownError()
+    }
+
+    const { message, status } = parsed
+    if (message && status && status >= 400 && status < 500) {
+      // Give api message back to user
+      throw new ClientRequestError(message)
+    }
+
+    // Log anything that's not due to bad input
+    context.log.error(decorateError(e), 'Error in login process.')
+
+    throw new UnknownError()
+  }
+
+  return {
```

Comment:
This is only reachable if the try block succeeded (cause the catch block will definitely throw something again). So I'd put this inside the try block for better comprehensibility.

Reply:
As we spoke, might be a matter of style, I find it more readable with less indentation and exactly after try/catch block.

Reply:
ok